diff -ruN spack-src/configure spack-src1/configure
--- spack-src/configure	2016-10-25 05:20:07.000000000 +0000
+++ spack-src1/configure	2020-09-15 19:47:10.039900399 +0000
@@ -5585,7 +5585,7 @@
 fi
 
 if test x"$with_mpi" = x"yes"; then
-  for ac_prog in mpiexec.hydra mpiexec mpirun poe runjob srun
+  for ac_prog in mpiexec.hydra mpiexec mpirun poe runjob srun aprun
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -5629,7 +5629,7 @@
 test -n "$MPI_BINARY" || MPI_BINARY="no"
 
   if test x"$MPI_BINARY" = x"no"; then
-    as_fn_error $? "Could not find either of the MPI binaries: mpiexec.hydra, mpiexec, mpirun, poe, runjob, srun" "$LINENO" 5
+    as_fn_error $? "Could not find either of the MPI binaries: mpiexec.hydra, mpiexec, mpirun, poe, runjob, srun, aprun" "$LINENO" 5
   fi
 fi
 
diff -ruN spack-src/configure.ac spack-src1/configure.ac
--- spack-src/configure.ac	2016-10-25 05:13:46.000000000 +0000
+++ spack-src1/configure.ac	2020-09-15 19:46:12.372578853 +0000
@@ -115,9 +115,9 @@
 
 dnl check which MPI binray invokes a MPI job
 if test x"$with_mpi" = x"yes"; then
-  AC_CHECK_PROGS([MPI_BINARY], [mpiexec.hydra mpiexec mpirun poe runjob srun], [no])
+  AC_CHECK_PROGS([MPI_BINARY], [mpiexec.hydra mpiexec mpirun poe runjob srun aprun], [no])
   if test x"$MPI_BINARY" = x"no"; then
-    AC_MSG_ERROR([Could not find either of the MPI binaries: mpiexec.hydra, mpiexec, mpirun, poe, runjob, srun])
+    AC_MSG_ERROR([Could not find either of the MPI binaries: mpiexec.hydra, mpiexec, mpirun, poe, runjob, srun, aprun])
   fi
 fi
 
